pkgs/top-level/all-packages.nix

```
@@ -11285,9 +11285,9 @@ in
 
   assimp = callPackage ../development/libraries/assimp { };
 
-  asio = asio_1_12;
+  asio = asio_1_16;
   asio_1_10 = callPackage ../development/libraries/asio/1.10.nix { };
-  asio_1_12 = callPackage ../development/libraries/asio/1.12.nix { };
+  asio_1_16 = callPackage ../development/libraries/asio/1.16.nix { };
```

Might as well drop the `asio_1_16` attribute. Out-of-tree expressions can always check asio.version if they need to.
```suggestion
  asio = callPackage ../development/libraries/asio/1.16.nix { };
```

---

Thank you :)

No worries, will squash commits and add colon.
Stand by for force-push.